### PR TITLE
Add zipAll support to IterableImpl

### PR DESCRIPTION
**Pull request by mduvall**
This commit adds `zipAll` support which zips over the iterables and uses the length of the longest iterable. There is a check to ensure that the invariant of the length of the longest iterable is always the size of the zipped value and a test to make sure leaves the holes in remaining zipped values are set to `undefined`.

### CODE REVIEW

File: src/Operations.js [hunk 1]
```diff
@@ -767,9 +774,11 @@ export function zipWithFactory(keyIter, zipper, iters) {
         steps = iterators.map(i => i.next());
         isDone = steps.some(s => s.done);
       }
+
```
Comment (mduvall) [inline src/Operations.js]: Added white space for breathing room, can remove if it is inconsistent in style.

File: src/Operations.js [hunk 2]
```diff
@@ -726,9 +726,15 @@ function maxCompare(comparator, a, b) {
 }
 
 
-export function zipWithFactory(keyIter, zipper, iters) {
+export function zipWithFactory(keyIter, zipper, iters, zipAll) {
   var zipSequence = makeSequence(keyIter);
-  zipSequence.size = new ArraySeq(iters).map(i => i.size).min();
+  var iterSizes = new ArraySeq(iters).map(i => i.size);
+  var zipSeqSize = !!zipAll ? iterSizes.max() : iterSizes.min();
+  if (zipAll) {
```
Comment (mduvall) [inline src/Operations.js]: Any opinions here? I was about to delve into the sequence's iterator but thought that might be much but went the route of normalizing the iterators lengths by just throwing on the `undefined`s at the end.

Comment (leebyron) [inline src/Operations.js]: Yeah this concat could be expensive. I think the logic to decide when to end the iteration should exist within the iterator.

File: src/Operations.js [hunk 3]
```diff
@@ -767,9 +774,11 @@ export function zipWithFactory(keyIter, zipper, iters) {
         steps = iterators.map(i => i.next());
         isDone = steps.some(s => s.done);
```
Comment (leebyron) [inline src/Operations.js]: I *think* that this is the critical point. a zipAll is done when *all* steps are done instead of *some* steps are done.

Comment (mduvall) [inline src/Operations.js]: Yeah that sounds right to me, let me see if I can get that to work.

**Comment by facebook-github-bot-4** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.

**Pull request closed without merging**